gcloud-java-pubsub/baseline/src/main/java/com/google/cloud/pubsub/spi/v1/PublisherApi.java

```diff
@@ -277,27 +277,46 @@ public void close() throws IOException {
   /**
    * Creates the given topic with the given name.
    *
+   * Sample code:
+   * <pre><code>
+   * try (PublisherApi publisherApi = PublisherApi.createWithDefaults()) {
+   *   String formattedName = PublisherApi.formatTopicName("[PROJECT]", "[TOPIC]");
+   *   Topic response = publisherApi.createTopic(formattedName);
+   * }
+   * </code></pre>
+   *
    * <!-- manual edit -->
    * <!-- end manual edit -->
    *
-   * @param name The name of the topic. It must have the format
+   * @param formattedName The name of the topic. It must have the format
    * `"projects/{project}/topics/{topic}"`. `{topic}` must start with a letter,
    * and contain only letters (`[A-Za-z]`), numbers (`[0-9]`), dashes (`-`),
    * underscores (`_`), periods (`.`), tildes (`~`), plus (`+`) or percent
    * signs (`%`). It must be between 3 and 255 characters in length, and it
    * must not start with `"goog"`.
    * @throws com.google.api.gax.grpc.ApiException if the remote call fails
    */
-  public final Topic createTopic(String name) {
-    Topic request = Topic.newBuilder().setName(name).build();
-
+  public final Topic createTopic(String formattedName) {
+    TOPIC_PATH_TEMPLATE.validate(formattedName);
```

Comment:
If code generation already treat the `formattedName` in a special way and validate it couldn't it replace it with 2 params (e.g. in this case project and topic)?

Also, because the generated message is part of this API I think you should find some consistent naming. I think it is a bit strange to call this one "formattedName" but when I get the *same* value from the Topic it is called "name".  It would have been easier if Topic.name was called Topic.fullyQualfiedName or had separate fields for project and name but mixing term that refer to the same value may be confusing.